### PR TITLE
Update NuGetAuthenticate to 1

### PR DESCRIPTION
**Pull request by Cosifne**
Update to NuGetAuthenticate1, otherwise CI would be blocked

### CODE REVIEW

**Comment by Cosifne** [comment]
Waiting a validation build now. https://dev.azure.com/dnceng/internal/_build/results?buildId=2361408&view=results

**Review by phil-allen-msft — APPROVED** [review]

File: eng/common/templates/job/job.yml [hunk 1]
```diff
@@ -136,7 +136,7 @@ jobs:
         condition: and(succeeded(), in(variables['_SignType'], 'real', 'test'), eq(variables['Agent.Os'], 'Windows_NT'))
 
   - ${{ if and(eq(parameters.runAsPublic, 'false'), eq(variables['System.TeamProject'], 'internal')) }}:
-    - task: NuGetAuthenticate@0
+    - task: NuGetAuthenticate@1
```
Comment (dibarbet) [inline eng/common/templates/job/job.yml:139]: These files (in eng/common) are managed by arcade and will get overwritten by the next arcade update.  We may already have an automated arcade update fixing this, or we need to get one from them.

**Review by dibarbet — CHANGES_REQUESTED** [review]

**Review by arkalyanms — APPROVED** [review]

File: eng/common/templates/job/job.yml (hunk 1 above)
Comment (Cosifne) [reply eng/common/templates/job/job.yml:139]: Let me see if the arcade updated is ignored somehwere

Comment (Cosifne) [reply eng/common/templates/job/job.yml:139]: Is this supposed to update the task? I don't see it's change this  https://github.com/dotnet/roslyn/pull/71604

**Comment by Cosifne** [comment]
Closed because synced with David that we want to arcade update these for us

**Pull request closed without merging**